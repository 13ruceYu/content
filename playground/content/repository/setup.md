---
title: Setup
description: Setup the repository locally.
---

## Install dependencies

```bash
yarn install
```

## Type support

We write most of the **@nuxt/content** code using [TypeScript](https://www.typescriptlang.org).

If you want to get full type support from Nuxt 3, you might want to run an initial build of `docs` directory.

Nuxt 3 generate its tsconfig.json on first build, so we need this local build to get full support for type aliases.

```bash
yarn build:docs
```
