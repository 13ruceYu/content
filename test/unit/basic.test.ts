import { assert, test, describe, expect, afterAll } from 'vitest'
import { setupTest } from '../setup'

describe('Basic tests', () => {
  const ctx = setupTest({
    fixture: 'basic',
    server: true
  })

  afterAll(ctx._destroy)

  test('Build', ctx._init, 0)

  test('List contents', async () => {
    const list = await ctx.fetch<Array<string>>('/api/_content/list')

    assert(list.length > 0)
    assert(list.includes('content:index.md'))

    // Ignored files should be listed
    assert(list.includes('content:.dot-ignored.md') === false, 'Ignored files with `.` should not be listed')
    assert(list.includes('content:-dash-ignored.md') === false, 'Ignored files with `-` should not be listed')
  })

  test('Get contents index', async () => {
    const index = await ctx.fetch<any>('/api/_content/get/content:index.md')

    expect(index).toHaveProperty('meta.mtime')
    expect(index).toHaveProperty('body')

    expect(index.body).toMatchSnapshot('basic-index-body')
  })

  test('Get ignored contents', async () => {
    const index = await ctx.fetch<any>('/api/_content/get/content:.ignored.md')

    expect(index).not.toHaveProperty('meta.mtime')
    expect(index).toMatchObject({})
    expect(index.body).toBeNull()
  })

  test('Get navigation', async () => {
    const list = await ctx.fetch<Array<string>>('/api/_content/navigation')

    expect(list).toMatchSnapshot('basic-navigation')
  })

  test('Get cats navigation', async () => {
    const list = await ctx.fetch<Array<string>>('/api/_content/navigation', {
      method: 'POST',
      body: {
        slug: '/cats'
      }
    })

    expect(list).toMatchSnapshot('basic-navigation-cats')
  })

  test('Get cats navigation', async () => {
    const list = await ctx.fetch<Array<string>>('/api/_content/navigation', {
      method: 'POST',
      body: {
        slug: '/dogs'
      }
    })

    expect(list).toMatchSnapshot('basic-navigation-dogs')
  })
})
