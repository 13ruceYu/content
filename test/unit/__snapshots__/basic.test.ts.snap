// Vitest Snapshot v1

exports[`Basic tests > Get cats navigation > basic-navigation-cats 1`] = `
[
  {
    "children": [
      {
        "draft": false,
        "id": "content:cats:index.md",
        "partial": false,
        "slug": "/cats",
        "title": "Cats list",
      },
      {
        "draft": false,
        "id": "content:cats:persian.md",
        "partial": false,
        "slug": "/cats/persian",
        "title": "Persian",
      },
      {
        "draft": false,
        "id": "content:cats:ragdoll.md",
        "partial": false,
        "slug": "/cats/ragdoll",
        "title": "Ragdoll",
      },
    ],
    "id": "content:cats:index.md",
    "slug": "/cats",
    "title": "Cats list",
  },
]
`;

exports[`Basic tests > Get cats navigation > basic-navigation-dogs 1`] = `
[
  {
    "children": [
      {
        "draft": false,
        "id": "content:dogs:bulldog.md",
        "partial": false,
        "slug": "/dogs/bulldog",
        "title": "Bulldog",
      },
      {
        "draft": false,
        "id": "content:dogs:German Shepherd.md",
        "partial": false,
        "slug": "/dogs/German Shepherd",
        "title": "German Shepherd",
      },
    ],
    "slug": "/dogs",
    "title": "Dogs",
  },
]
`;

exports[`Basic tests > Get contents index > basic-index-body 1`] = `
{
  "children": [
    {
      "children": [
        {
          "type": "text",
          "value": "Index",
        },
      ],
      "props": {
        "class": "d-heading-title",
        "id": "index",
      },
      "tag": "prose-h1",
      "type": "element",
    },
  ],
  "toc": {
    "depth": 2,
    "links": [],
    "searchDepth": 2,
    "title": "",
  },
  "type": "root",
}
`;

exports[`Basic tests > Get navigation > basic-navigation 1`] = `
[
  {
    "children": [
      {
        "draft": false,
        "id": "content:index.md",
        "partial": false,
        "slug": "/",
        "title": "Index",
      },
    ],
    "id": "content:index.md",
    "slug": "/",
    "title": "Index",
  },
  {
    "children": [
      {
        "draft": false,
        "id": "content:dogs:bulldog.md",
        "partial": false,
        "slug": "/dogs/bulldog",
        "title": "Bulldog",
      },
      {
        "draft": false,
        "id": "content:dogs:German Shepherd.md",
        "partial": false,
        "slug": "/dogs/German Shepherd",
        "title": "German Shepherd",
      },
    ],
    "slug": "/dogs",
    "title": "Dogs",
  },
  {
    "children": [
      {
        "draft": false,
        "id": "content:cats:index.md",
        "partial": false,
        "slug": "/cats",
        "title": "Cats list",
      },
      {
        "draft": false,
        "id": "content:cats:persian.md",
        "partial": false,
        "slug": "/cats/persian",
        "title": "Persian",
      },
      {
        "draft": false,
        "id": "content:cats:ragdoll.md",
        "partial": false,
        "slug": "/cats/ragdoll",
        "title": "Ragdoll",
      },
    ],
    "id": "content:cats:index.md",
    "slug": "/cats",
    "title": "Cats list",
  },
]
`;
